community/kernel/src/main/java/org/neo4j/storageengine/api/StorageEngine.java

```diff
@@ -113,6 +114,14 @@ void createCommands(
      */
     void prepareForRecoveryRequired();
 
+    /**
+     * As part of the simplified counts store. This will initialize the count store in the storage engine
+     * when doing recovery.
+     */
+    void initFromSnapshot( CountsSnapshot snapshot );
```

Comment:
Where to add that instead? Note that a snapshot is pretty general (a long and a map) and unrelated to storage implementation details.